src/git-directory-helper.ts

```diff
@@ -35,6 +35,14 @@ export async function prepareExistingDirectory(
       path.join(repositoryPath, '.git', 'index.lock'),
       path.join(repositoryPath, '.git', 'shallow.lock')
     ]
+    const lockDir = path.join(repositoryPath, '.git')
+    
+    for (const file of await fsHelper.readdirRecursive(lockDir)) {
+      if (file.endsWith('index.lock') || file.endsWith('shallow.lock')) {
+        lockPaths.push(path.join(lockDir, file))
```

Comment:
maybe you tested this and it works but from my glance at the logic here it seems like there may be a bug? `readdirRecursive` seems to output a list of filepath strings that already have `lockDir` as a base, so `file` here should already have `lockDir` prepended

Reply:
to that point, probably leave a docstring 
```typescript
/**
 * This function searches a given directory and returns a list of file paths giving all files in that directory.
 * The file paths are all prefixed with `dir`.
 *  
 * @param dir The directory to 
 */
export async function readdirRecursive(dir: string) { ... }

Reply:
Yep you're entirely correct. (Cascade also agreed). Going to test now